File: sdk/python/src/tinyhumans/api/announcements.py
from __future__ import annotations

from typing import Any

from ..http import Json
from ._base import ApiNamespace

__all__ = ["AnnouncementsApi"]


class AnnouncementsApi(ApiNamespace):
    """Active in-app announcements for the signed-in user."""

    def get_latest_announcements(self, **kwargs: Any) -> Json:
        """Get the latest active announcement for the signed-in user."""
        return self._http.get("/announcements/latest", **kwargs)

File: src/api/announcements.rs
//! Active in-app announcements for the signed-in user.

use reqwest::Method;
use serde_json::Value;

use crate::{Error, HttpClient};

/// Typed client for the `/announcements/*` routes.
pub struct AnnouncementsApi<'a> {
    http: &'a HttpClient,
}

impl<'a> AnnouncementsApi<'a> {
    pub fn new(http: &'a HttpClient) -> Self {
        Self { http }
    }

    /// Get the latest active announcement for the signed-in user.
    pub async fn get_latest_announcements(&self) -> Result<Value, Error> {
        self.http
            .send(Method::GET, "/announcements/latest", &[], None, true)
            .await
    }
}
